File: LeetCode/Simulation/JS/Leetcode1455-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.js
var isPrefixOfWord = function(sentence, searchWord) {
    let [ans,rec,idx] = [-1,sentence.split(' '),0];
    let check = function(s,p) {
        if(p.length > s.length) {
            return false;
        }
        for(let i = 0;i < p.length;i++) {
            if(s[i] != p[i]) return false;
        }
        return true;
    }
    for(const s of rec) {
        idx++;
        if(check(s,searchWord)) {
            ans = idx;
            break;
        }
    }
    return ans; 
};

File: LeetCode/Simulation/Python/Leetcode1455-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        rec,idx,ans = sentence.split(' '),0,-1
        print(rec)
        def check(s,p):
            if len(p) > len(s): return False
            for i in range(len(p)):
                if s[i] != p[i]: return False
            return True
        for s in rec:
            idx += 1
            if check(s,searchWord):
                ans = idx
                break
        return ans